AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp

```diff
@@ -350,7 +352,7 @@ WebSocketReceiveResult *WebSocketClient::receive(char *msgBuffer, int size)
             readChar(&c);
 
             int shift = (7 - i) * 8 > 32 ? 32 : (7 - i) * 8;
-            payloadLength += (c << shift);
+            payloadLength += c << (unsigned(shift));
         }

```

Comment:
readChar(&c);  // 0
readChar(&c);  // 0
readChar(&c);  // 0
readChar(&c);  // 0
readChar(&c);
payloadLength = c << 24;
readChar(&c);
payloadLength += c << 16;
readChar(&c);
payloadLength += c << 8;
readChar(&c);
payloadLength += c;

Reply:
Updated, please check it.
as of now, there is no cpp error, and it works well